src/main/java/sx/blah/discord/api/internal/DiscordWS.java

```diff
@@ -593,8 +593,9 @@ private void ready(JsonElement eventObject) {
 
 			Discord4J.LOGGER.debug(LogMarkers.WEBSOCKET, "Logged in as {} (ID {}).", client.ourUser.getName(), client.ourUser.getID());
 
-			if(!this.async) {
+			if(this.async) {
 				this.isReady = true;
```

Comment:
I think this line and the next should be generalized into a private setReady() method or something like that. As it is the same as in the logic on guildCreate() which could cause problems should the loading logic need to be changed as it is possible one portion of the code won't be updated due to basic human error.